content/blog/2017-11-28-fragments-in-react-v16.2.md

```diff
@@ -0,0 +1,181 @@
+
+```jsx
+render() {
+  return [
+    <span key="first-span">I am</span>,
+    <span key="second-span">static</span>
+  ];
+}
+```
+
+However, the keys are only necessary for React's reconciliation algorithm when child elements are re-arranged and React needs to figure out which item in one render corresponds to in a subsequent render. If the array is static, then keys aren't necessary and only add extra noise to the code. So, let's omit them:
+
+```jsx
+render() {
+  return [
+    <span>I am</span>,
+    <span>static</span>
+  ];
+}
+```
+
+But React gives a key warning with the above code! This is because React doesn't know whether the array items can ever rearrange. With fragment syntax, it is clear that the code's intent is to render a static list of children, and not a developer mistake:
+
+```jsx
+render() {
+  return (
+    <>
+      <span>I am</span>
+      <span>static</span>
+    </>
+  );
+}
+```
+
+#### The Syntax
+
+Inspired by some prior art such as the `XMLList() <></>` constructor in [E4X](https://developer.mozilla.org/en-US/docs/Archive/Web/E4X/E4X_for_templating) and the existing `<></>` syntax in [Reason React](https://reasonml.github.io/reason-react/), the React team concluded on having a convenient syntax for JSX fragments that looks like `<>...</>`. This captures the idea of a fragment: It is a pair of empty tags which enforces the notion that it won't render to an actual element in the DOM.
+
+The `<></>` syntax also makes it easier to refactor since it looks just like a JSX element. For example, consider swapping the parents in the following code snippets:
+
+```jsx
+// with a parent wrapper
+<div>
+  <span>foo</span>
+  bar
+  <span>baz</span>
+</div>
+
+// with arrays (where commas are necessary and text needs to be surrounded by quotes)
+[
+  <span>foo</span>,
+  'bar',
+  <span>baz</span>
+]
+
+// with jsx fragment syntax
+<>
+  <span>foo</span>
+  bar
+  <span>baz</span>
+</>
+```
+
+#### The Export
+
+After the release of React 16, people in the React community have created solutions in order to address the problem of static lists, such as [react-aux](https://github.com/gajus/react-aux). For `Fragment`s however, the React team has decided to take ownership of the export, since there are some semantics specific to Fragments that aren't possible without core library changes. These changes have some performance wins and make reconciliation a bit more intuitive.
+
+[For the curious, you can read more about that here.](https://github.com/facebook/react/pull/10783)
+
+Still, we are very thankful to the open source contributors who worked on solving this in the ecosystem.
+
+Additionally, there is a [pragma option available in babel-plugin-transform-react-jsx](https://github.com/babel/babel/tree/master/packages/babel-plugin-transform-react-jsx#pragmafrag) that allows the Babel compiler to de-sugar the `<>...</>` syntax to a custom identifier.
```

Comment:
Maybe replace "Additionally" with "If you use JSX with something other than React, " to better explain the context of this sentence.